tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchCMISTests.java

```diff
@@ -172,6 +174,41 @@ public void checkInSyntax()
         searchQueryService.expectResultsFromQuery(query, user1, FILE_0_NAME, FILE_1_NAME);
     }
 
+    @Test (groups = TestGroup.SEARCH)
+    public void negative_basicCMISQuery_missingFrom()
+    {
+        // note: ideally 400 but currently 500 :-(
+
+        SearchRequest query1 = req("cmis", "SELECT *");
+        searchQueryService.expectErrorFromQuery(query1, user1, HttpStatus.INTERNAL_SERVER_ERROR, "expecting FROM");
+
+        SearchRequest query2 = req("cmis", "SELECT * FROM");
+        searchQueryService.expectErrorFromQuery(query2, user1, HttpStatus.INTERNAL_SERVER_ERROR, "no viable alternative at input");
+    }
+
+    @Test (groups = TestGroup.SEARCH)
+    public void negative_basicCMISQuery_invalidType()
+    {
+        // note: ideally 400 but currently 500 :-(
+        SearchRequest query = req("SELECT * FROM cmis:unknown");
+        searchQueryService.expectErrorFromQuery(query, user1, HttpStatus.INTERNAL_SERVER_ERROR, "Unknown property");
```

Comment:
This error is pretty confusing. Does it mention `cmis:unknown` in the full error message?

Reply:
Yes, this one is odd. It is also different to equivalent test at lower-level which gives a more useful "[Type is unsupported in query](https://github.com/Alfresco/alfresco-enterprise-repo/pull/809/files#diff-2466e672e08f8f55c9f5273d5f2a5555f148ce057aaf137da65bbd7df917f35cR95)" (note: private link)

TBC, but we may find this changes (gets fixed) as part of ACS-2920.

Reply:
I've updated the error messages to match currently what is returned. 

We may decide to revisit and improve, eg. if different from AFTS query lang &/or using CMIS query lang with Solr.